src/libcore/macros.rs

```diff
@@ -15,7 +15,8 @@ macro_rules! panic {
         panic!("explicit panic")
     );
     ($msg:expr) => ({
-        static _MSG_FILE_LINE: (&'static str, &'static str, usize) = ($msg, file!(), line!());
+        static _MSG_FILE_LINE: (&'static str, &'static str, usize) =
```

Comment:
Well.. still hitting the stage0 assertion in LLVM when using `#[cfg(stage0)]`.  I've added more details (backtrace of the stage0) in #19284.  I'm afraid I need help from someone more familiar with the compiler.